src/fondant/compiler.py

```diff
@@ -183,3 +184,61 @@ def _generate_spec(self, pipeline: Pipeline, extra_volumes: list) -> dict:
             "version": "3.8",
             "services": services,
         }
+
+
+class KubeFlowCompiler(Compiler):
+    """Compiler that creates a Kubeflow pipeline spec from a pipeline."""
+
+    def _resolve_imports(self):
+        """Resolve imports for the Kubeflow compiler."""
+        try:
+            import kfp
+            from kfp import dsl
+        except ImportError:
+            raise ImportError(
+                "You need to install kfp to use the Kubeflow compiler, "
+                / "you can install it with `poetry install --extras kfp`"
+            )
+
+    def compile(
+        self,
+        pipeline: Pipeline,
```

Comment:
I think the `Pipeline` is the unifying interface between all runners right? if so, would add it to the abstract class